### PR TITLE
Implement chunkify

### LINKED ISSUE #9: Create chunkify
large orders may cause issues with telegram message length limit

Create chunkify function (string message, int max_chunk_size) -> list[string]
max_chunk_size for telegram I believe is 4000 chars
Attempts to split at line breaks, if not possible then splits at spaces
Export as function first, if we have time can look into making this an npm node

Refer to https://github.com/Nukesor/ultimate-poll-bot/blob/main/pollbot/helper/text.py
Add additional checks that if (wack edge case) there are lines of > c*max_chunk_size, then that large line gets split by space (or worst case just hard split)
Write tests

### PR DESCRIPTION
**Pull request by vigonometry**

### CODE REVIEW

File: test/chunkify/chunkify_test.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    /**
+     * Tests whether exception is thrown when invalid/negative CHAR_LIMIT are passed into chunkify
+     */
+
+    it('should throw error when maxChars is less than 1', function(done) {
+        assert.throws(
+            () => {
+                chunkify(EMPTY_STRING, INVALID_CHAR_LIMIT);
+                chunkify(EMPTY_STRING, NEGATIVE_CHAR_LIMIT);
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js]: does this line get executed?

Comment (nicktohzyu) [reply test/chunkify/chunkify_test.js]: no need for abstracting the constants here. They're trivial and only used once

Comment (nicktohzyu) [reply test/chunkify/chunkify_test.js]: test with non-empty string too

File: test/chunkify/chunkify_test.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            }
+        );
+        done();
+    });
+
+    /**
+     * Tests whether small valid messages passed in as text files containing CHAR_LIMIT characters is un-modified.
+     */
+
+    it('should not modify valid small messages passed in as text files', function(done) {
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js]: To me, this implies that we're passing the message as a text file to chunkify

File: test/chunkify/chunkify_test.js (hunk 1 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js]: Bug fixed where line does not get executed.
Abstraction of constants removed.
Test with non-empty string and invalid chars added.

File: test/chunkify/chunkify_test.js (hunk 2 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js]: Updated documentation for said test case

File: test/chunkify/chunkify_test.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    /**
+     * Tests whether exception is thrown when invalid/negative CHAR_LIMIT are passed into chunkify
+     */
+
+    it('should throw error when maxChars is less than 1', function(done) {
+        const VALID_SMALL_TEST_MESSAGE = randomTextGenerator(SMALL_CHAR_LIMIT);
+        assert.throws(
+            () => {
+                chunkify(VALID_SMALL_TEST_MESSAGE, Math.floor((Math.random() * -10)));
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js]: You may use random to have a strictly negative number. It's also fine to hardcode the number.
Explicitly test 0 as well

Comment (nicktohzyu) [reply test/chunkify/chunkify_test.js]: Test with empty string and also non-empty string

File: test/chunkify/chunkify_test.js [hunk 4]
```diff
@@ -122,4 +120,20 @@ describe('Chunkify', function () {
...
+    /**
+     * Tests whether exception is thrown when invalid/negative CHAR_LIMIT and a text file converted to a String are passed into chunkify
+     */
+
+    it('should throw error when maxChars is less than 1', function(done) {
+        const testFilePath = "test/chunkify/testInputSmall.txt";
+        const VALID_SMALL_TEST_MESSAGE = readTextFile(testFilePath);
+        assert.throws(
+            () => {
+                chunkify(VALID_SMALL_TEST_MESSAGE, 0);
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js]: could you group this together with the other invalid maxChars test

File: test/chunkify/chunkify_test.js (hunk 3 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js]: Added new assert statements a to resolve this

File: test/chunkify/chunkify_test.js (hunk 4 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js]: Grouped in new commit

File: test/chunkify/chunkify_test.js [hunk 5]
```diff
@@ -23,70 +23,61 @@ describe('Chunkify', function () {
     });
 
     /**
-     * Tests whether valid message within a specified CHAR_LIMIT is un-modified.
+     * Tests whether randomized valid message within a specified CHAR_LIMIT is un-modified.
      */
 
-    it('should not modify valid simple messages', function(done) {
+    it('should not modify valid simple messages (randomized input)', function(done) {
         const actualChunkifiedMessage = chunkify("Valid Message", TELEGRAM_CHAR_LIMIT);
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js:30]: This input isn't randomized?

File: test/chunkify/chunkify_test.js [hunk 6]
```diff
@@ -23,70 +23,61 @@ describe('Chunkify', function () {
...
 
     /**
-     * Tests whether large messages containing number of characters greater than CHAR_LIMIT characters are modified appropriately
-     * (i.e. no Empty Strings within output).
+     * Tests whether randomized, large messages containing number of characters greater than CHAR_LIMIT characters
+     * are modified appropriately (i.e. no Empty Strings within output).
      */
 
-    it('should split large messages in order of linebreaks, whitespace and characters where appropriate', function(done) {
+    it('should split large messages in order of linebreaks, whitespace and chars where appropriate (randomized input)', function(done) {
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js]: This test doesn't check for the order though! It only verifies that the length is within certain bounds. I might write "should return an array of strings with strictly positive size less than the char limit"

File: test/chunkify/chunkify_test.js [hunk 7]
```diff
@@ -122,18 +114,37 @@ describe('Chunkify', function () {
...
     /**
-     * Tests whether exception is thrown when invalid/negative CHAR_LIMIT and a text file converted to a String are passed into chunkify
+     * Tests whether exception is thrown when invalid/negative CHAR_LIMIT are passed into chunkify
      */
 
     it('should throw error when maxChars is less than 1', function(done) {
         const testFilePath = "test/chunkify/testInputSmall.txt";
-        const VALID_SMALL_TEST_MESSAGE = readTextFile(testFilePath);
+        const FIXED_VALID_SMALL_TEST_MESSAGE = readTextFile(testFilePath);
+        const RANDOM_VALID_SMALL_TEST_MESSAGE = randomTextGenerator(SMALL_CHAR_LIMIT);
```
Comment (nicktohzyu) [inline test/chunkify/chunkify_test.js:108]: test with empty string as well please

**Review by nicktohzyu — CHANGES_REQUESTED** [review]
Could you please group them such that the same test (random + fixed) are together

File: test/chunkify/chunkify_test.js (hunk 5 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js:30]: Updated

File: test/chunkify/chunkify_test.js (hunk 6 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js]: Updated per specifications

File: test/chunkify/chunkify_test.js (hunk 7 above)
Comment (vigonometry) [reply test/chunkify/chunkify_test.js:108]: Updated per specifications

File: test/chunkify/randomTextGenerator.js [hunk 8]
```diff
@@ -15,4 +19,19 @@ function randomTextGenerator(length) {
...
 
-exports.randomTextGenerator = randomTextGenerator;
\ No newline at end of file
+function randomTextFileGenerator(length){
+    let result = '';
+    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789\n\r ';
+    const charactersLength = characters.length;
+    for (let i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
```
Comment (sheehui) [inline test/chunkify/randomTextGenerator.js]: This portion is a duplicated code, can consider calling
```javascript
const result = randomTextGenerator(length)
```

**Review by sheehui — COMMENTED** [review]
I noticed that `test/chunkify` and `test/` both contain `readTextFile.js` and `randomTextGenerator.js`. There should only be one instance of each file if they contain the same functions! If only chunkify uses it, can remove the ones in `test/`, if there is a need for these functions for other test files in the future, we can move it to a `commons` folder.

File: helper/chunkify.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    let chunks = [];
+
+    if (message.length <= maxChars){
+        chunks.push(message);
+        return chunks;
+    }
+
+    const firstSection = message.slice(0, maxChars);
+
+    const nextlineIdx = firstSection.lastIndexOf('\n');
```
Comment (sheehui) [inline helper/chunkify.js]: small nitpick, camel casing => `nextLineIdx`

**Review by nicktohzyu — APPROVED** [review]
